### PR TITLE
Fix rename on a union

### LINKED ISSUE #715: `Schema.rename` on a union and inline fragments
Given the following schema:
```scala
sealed trait Union
object Union {
  case class Child(field: String) extends Union
}

case class Obj(
  union: Union,
)
case class Query(
  test: Obj,
)

object Schema {
  implicit val schemaUnionChild: Schema[Any, Union.Child] = Schema.gen[Union.Child].rename("UnionChild")
  implicit val schemaTestUnion: Schema[Any, Union] = Schema.gen[Union]
  implicit val schemaQuery: Schema[Any, Query] = Schema.gen[Query]
}
```

A query:
```graphql
query {
  test {
    union {
      __typename
      ... on UnionChild {
        field
      }
    }
  }
}
```

Produces the expected result:
```json
{
  "test": {
    "union": {
      "__typename": "UnionChild",
      "field": "f"
    }
  }
}
```

Renaming the `Union` using the `rename`  method on `Schema`:
```scala
implicit val schemaTestUnion: Schema[Any, Union] = Schema.gen[Union].rename("UnionRenamed")
```
gives the following:
```json
{
  "test": {
    "union": {
      "__typename": "UnionRenamed"
    }
  }
}
```

Using the `@GQLName` annotation produces the expected result.

### PR DESCRIPTION
**Pull request by ghostdogpr**
Closes #715

### CODE REVIEW

File: core/src/test/scala/caliban/execution/ExecutionSpec.scala [hunk 1]
```diff
@@ -450,6 +450,70 @@ object ExecutionSpec extends DefaultRunnableSpec {
...
+                   ... on UnionChild {
+                     field
+                   }
+                 }
+               }
+             }""")
+
+        assertM(interpreter.flatMap(_.execute(query)).map(_.data.toString))(
+          equalTo("""{"test":{"union":{"__typename":"UnionChild","field":"f"}}}""".stripMargin)
+        )
```
Comment (paulpdaniels) [inline core/src/test/scala/caliban/execution/ExecutionSpec.scala:515]: I think should also verify that the parent got renamed.

Comment (ghostdogpr) [reply core/src/test/scala/caliban/execution/ExecutionSpec.scala:515]: The parent will never be returned by `__typename`, right?

Comment (paulpdaniels) [reply core/src/test/scala/caliban/execution/ExecutionSpec.scala:515]: It would show up in the introspection / schema though I think.

Comment (ghostdogpr) [reply core/src/test/scala/caliban/execution/ExecutionSpec.scala:515]: Right. This isn't affected by this PR though so I'll merge this one (finally 😄)

**Pull request merged**